### PR TITLE
Chore: Improves success story redirects.

### LINKED ISSUE #3851: Enhancement: Fix redirection on Success Stories section
Complete the following REQUIRED checkboxes:

- [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
 - [x] The title of this issue follows the brief description of request format, e.g. Add dark mode to website
 
The following checkbox is OPTIONAL:

 - [ ] I would like to be assigned this issue to work on it
1. Description of the Feature Request:

On the main landing page, there is “Success Stories” section. This section has students’ icons, names, the beginning sentence of the stories, and a button to go to the static page of the success stories.
When clicking on their names, each leads to the student’s social media link if it is specified on this [file](https://github.com/TheOdinProject/theodinproject/blob/main/db/seeds/success_stories.rb). However, if the student does not specify anything, then the click takes it to the very top of the landing page. It is the same response on their [static page](https://github.com/TheOdinProject/theodinproject/blob/main/app/views/static_pages/success_stories.html.erb) as it is rendered. I believe it will make more sense if clicking anywhere on this section can lead to the success stories page instead of going to different websites or the front of the landing page.

Screenshot 2023-06-14 140654
2. Acceptance Criteria:

Click on the card on the Success Stories section on the landing page and it takes you to the static page
On the static page, click on the user without the social media link and it does nothing (maybe remove '#' on the [file](https://github.com/TheOdinProject/theodinproject/blob/main/db/seeds/success_stories.rb))

3. Additional Information:

### PR DESCRIPTION
**Pull request by ZachBaird**
## Because
Success stories current always link to *somewhere*. Ostensibly, this would be the social_media_link associated with the success story, but not all success stories had a link. This created a confusing UX experience.

## This PR

- Modifies the success story section on the home page to point to the specific story on the success_stories page.
- On the success_stories page, the user name and a link.svg are presented as links to the user's social media if their link is present.
- A normal `p` tag is rendered if there is no social media link present.
- Allows the CardComponent to take in an id.

## Issue
Closes #3851

## Pull Request Requirements
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behavior
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] I have verified all tests and linters pass after making these changes.
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If applicable, this PR includes new or updated automated tests

### CODE REVIEW

File: app/views/static_pages/success_stories.html.erb [hunk 1]
```diff
@@ -6,14 +6,32 @@
...
             </div>
             <div class="flex flex-col text-center sm:text-left">
-              <p class="font-bold"><%= link_to story.student_name, story.social_media_link, class: 'underline hover:no-underline text-gray-600 hover:text-gray-800 dark:text-gray-200 dark:hover:text-gray-300' %></p>
+              <div class="flex flex-row gap-2">
+                <% if story.social_media_link.present? %>
+                  <p class="font-bold"><%= link_to story.student_name, story.social_media_link, class: 'underline hover:no-underline text-gray-600 hover:text-gray-800 dark:text-gray-200 dark:hover:text-gray-300' %></p>
+                  <%= link_to story.social_media_link do %>
+                    <%= inline_svg_tag(
+                          'icons/link.svg',
+                          class: 'h-6 w-6 text-gray-400 dark:text-gray-500 hover:text-gray-800 dark:hover:text-gray-300',
```
Comment (KevinMulhern) [inline app/views/static_pages/success_stories.html.erb:22]: Nit: proportionally to the name and description, this looks just _slightly_ too large for me. 

Whats your thoughts on using something around the w-4 range instead?
```suggestion
                          class: 'h-4 w-4 text-gray-400 dark:text-gray-500 hover:text-gray-800 dark:hover:text-gray-300',
```

Screenshot 2023-09-08 at 17 45 59

File: app/views/static_pages/success_stories.html.erb [hunk 2]
```diff
@@ -6,14 +6,32 @@
...
-      <%= render CardComponent.new(classes: 'mb-10') do |card| %>
+      <%= render CardComponent.new(classes: 'mb-10', id: dom_id(story)) do |card| %>
         <% card.with_header do %>
           <div class="flex flex-col sm:flex-row items-center space-y-2 sm:space-y-0 sm:space-x-4">
             <div class="w-20 h-20 rounded-full overflow-hidden">
               <%= image_tag "success_stories/#{story.avatar_path_name}", alt: '', class: 'w-full shadow-inner aspect-auto' %>
             </div>
             <div class="flex flex-col text-center sm:text-left">
-              <p class="font-bold"><%= link_to story.student_name, story.social_media_link, class: 'underline hover:no-underline text-gray-600 hover:text-gray-800 dark:text-gray-200 dark:hover:text-gray-300' %></p>
+              <div class="flex flex-row gap-2">
```
Comment (KevinMulhern) [inline app/views/static_pages/success_stories.html.erb:16]: Related the previous suggestion - If we have a smaller icon, it'll look best of centered with the name.

```suggestion
              <div class="flex flex-row gap-2 items-center">
```

**Review by KevinMulhern — COMMENTED** [review]
Great stuff @ZachBaird 🔥  thanks for doing this so quickly!

Just a couple of very minor styling suggestions from me, but this is all first-rate!

**Comment by ZachBaird** [comment]
Mint suggestions. Committing them now.

**Review by KevinMulhern — APPROVED** [review]
Love your work mate 👏

**Pull request merged**